### PR TITLE
include and use clap only if needed

### PR DESCRIPTION
**Pull request by dmitris**
Follow-up on https://github.com/aya-rs/aya-template/pull/67 to make clippy completely happy on the generated code.
/cc @vadorovsky

### CODE REVIEW

File: {{project-name}}/src/main.rs [hunk 1]
```diff
@@ -39,7 +39,11 @@ use aya::programs::SocketFilter;
 use aya::programs::RawTracePoint;
 {%- endcase %}
 use aya_log::BpfLogger;
+{% case program_type %}
+{%- when
+    "xdp", "classifier", "sock_ops", "cgroup_skb", "cgroup_sysctl", "cgroup_sockopt", "uprobe", "uretprobe" -%}
```
Comment (vadorovsky) [inline {{project-name}}/src/main.rs:44]: Before I approve this - are you sure that there is no way to define this list once as a variable and use it later? It's kinda ugly that we copy that list over.

Comment (dmitris) [reply {{project-name}}/src/main.rs:44]: I know but I couldn't find anything despite looking through the docs.  The templating engine doesn't seem to allow setting of "general" variables - only the placeholders that are tied to prompts, and that's not what we want.  As mentioned on https://discord.com/channels/855676609003651072/855676609003651075/1061962945850003496 I tried
> to put` {% assign program_type_with_opts = "xdp, classifier, sock_ops, cgroup_skb, cgroup_sysctl, cgroup_sockopt, uprobe, uretprobe" | split: ", " %}` in the {{project-name}}/src/main.rs - but cargo generate then doesn't generate <project-name>/src at all (with no error) - apparently it doesn't like that syntax...

Comment (dmitris) [reply {{project-name}}/src/main.rs:44]: maybe we should at least add a `DEVEL.md` file to highlight the pitfalls such as the need to update the Opt/clap related lists of programs in sync?

Comment (dmitris) [reply {{project-name}}/src/main.rs]: updated the PR to use the template's `assign` - currently repeated in two files, haven't found (yet) a way to share the assigned variable between the files

**Comment by dmitris** [comment]
to test:
`rm -fr kprobetest && cargo generate --name kprobetest -d program_type=kprobe -d kprobe=tcp_connect -g https://github.com/dmitris/aya-template -b fix-clippy`
`rm -fr xdptest && cargo generate --name xdptest -d program_type=xdp -g https://github.com/dmitris/aya-template -b fix-clippy`

**Review by vadorovsky — COMMENTED** [review]
This branch doesn't seem to work with simple `cargo generate --git` usage:

```
aya-examples/dmitris on  main [?] 
❯ cargo generate --git https://github.com/dmitris/aya-template -b fix-clippy --init
🤷   Project Name : dmitris
🔧   Destination: /home/vadorovsky/repos/aya-examples/dmitris ...
🔧   Generating template ...
✔ 🤷   Which type of eBPF program? · classifier
✔ 🤷   Attach direction? · Ingress
[...]

aya-examples/dmitris on  main [?] via  v1.66.1 took 8s 
❯ RUST_LOG=info cargo xtask run -- -i wlp5s0
[...]
   Compiling dmitris v0.1.0 (/home/vadorovsky/repos/aya-examples/dmitris/dmitris)
error[E0425]: cannot find value `opt` in this scope
  --> dmitris/src/main.rs:29:35
   |
29 |     let _ = tc::qdisc_add_clsact(&opt.iface);
   |                                   ^^^ not found in this scope

error[E0425]: cannot find value `opt` in this scope
  --> dmitris/src/main.rs:32:21
   |
32 |     program.attach(&opt.iface, TcAttachType::Ingress)?;
   |                     ^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
error: could not compile `dmitris` due to 2 previous errors
thread 'main' panicked at 'assertion failed: status.success()', xtask/src/run.rs:34:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

It seems like using `cargo generate --git` doesn't honor the `[hooks]` config from `cargo-generate.toml`. I have no idea why, but we need to find out before merging this.

**Comment by vadorovsky** [comment]
Same for

```
cargo generate --name xdptest -d program_type=xdp -g https://github.com/dmitris/aya-template -b fix-clippy
```

It doesn't work

**Review by vadorovsky — CHANGES_REQUESTED** [review]
OK, I got cargo-generate working, but whitespaces are wrong for programs with clap:

dmitris-aya-template-1

I don't have any quick fix in mind, but you probably need to adjust `%-` or `-%` around the `Opt` struct and find the combination which generates one empty line between imports, the struct and the main function, both and without clap.

**Comment by dmitris** [comment]
@vadorovsky  could you try now, please?

**Review by vadorovsky — APPROVED** [review]
Nice, it works great now. Thanks a lot!

**Pull request merged**